pkg/splunk/enterprise/util.go

```diff
@@ -292,3 +294,236 @@ func DeleteOwnerReferencesForS3SecretObjects(client splcommon.ControllerClient,
 
 	return err
 }
+
+// checkIfAnAppIsActive checks if the App is listed as part of the AppSrc listing
+func checkIfAnAppIsActiveOnRemoteStore(appName string, list []*splclient.RemoteObject) bool {
+	for i := range list {
+		if strings.HasSuffix(*list[i].Key, appName) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// checkIfAppSrcExistsWithRemoteListing checks if a given AppSrc is part of the remote listing
+func checkIfAppSrcExistsWithRemoteListing(appSrc string, remoteObjListingMap map[string]splclient.S3Response) bool {
+	if _, ok := remoteObjListingMap[appSrc]; ok {
+		return true
+	}
+
+	return false
+}
+
+// changeAppSrcDeployInfoStatus sets the new status to all the apps in an AppSrc if the given repo state and deploy status matches
+// primarly used in Phase-3
```

Comment:
How will this be used in Phase 3?

Reply:
high-level idea is, when we ask the sidecar with App source info(s3 info, and secrets), and when the deployment work is done, we change all the apps under that app source to the complete state. We still have some unknowns at this time how we handle a single app failure case, otherwise, we are handling the App source as a single unit.